File: frontend/types/bank.ts
export type BankInfoType = {
  id: number;
  name: string;
  address: string;
  nip: string;
  krs: string;
  capitalization: string;
  created_at: string;
  updated_at: string;
};

export type AverageDepositRate = {
  normal_interest_rate__avg: number;
};

export type BankDepositAverageRate = {
  bank__name: string;
  average_rate: number;
};

File: backend/finances/models/bank.py
from django.db import models


class Bank(models.Model):
    name = models.CharField(max_length=1024)
    address = models.CharField(max_length=1024, blank=True, default="")
    nip = models.CharField(max_length=13, blank=True, default="")
    krs = models.CharField(max_length=10, blank=True, default="")
    capitalization = models.DecimalField(
        max_digits=12, decimal_places=2, blank=True, null=True
    )
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        constraints = [
            models.UniqueConstraint(fields=["name"], name="unique_bank_name"),
        ]

    def __str__(self):
        return self.name
